fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -5475,20 +5521,26 @@ public void save(DataOutputStream dos) throws IOException, SRMetaBlockException
                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
         int totalTableNum = 0;
         for (Database database : idToDbNormal.values()) {
-            totalTableNum += database.getTableNumber();
+            totalTableNum += database.getTableNumber() + database.getTemporaryTableNumber();
         }
         int cnt = 1 + idToDbNormal.size() + idToDbNormal.size() /* record database table size */ + totalTableNum + 1;
-
         SRMetaBlockWriter writer = new SRMetaBlockWriter(dos, SRMetaBlockID.LOCAL_META_STORE, cnt);
 
         writer.writeJson(idToDbNormal.size());
         for (Database database : idToDbNormal.values()) {
             writer.writeJson(database);
-            writer.writeJson(database.getTables().size());
+            int totalTableNumber = database.getTables().size() + database.getTemporaryTables().size();
+            writer.writeJson(totalTableNumber);
             List<Table> tables = database.getTables();
             for (Table table : tables) {
                 writer.writeJson(table);
             }
+
+            List<Table> tempTables = database.getTemporaryTables();
+            for (Table table : tempTables) {
+                writer.writeJson(table);
```

Comment:
test version rollback?

Reply:
> test version rollback?

This change is not forward compatible. Before rolling back,  need to ensure that there are no temporary tables in the cluster.